hikari/webhooks.py

```diff
@@ -59,96 +60,48 @@ class WebhookType(int, enums.Enum):
-    """The guild ID of the webhook."""
-
-    channel_id: snowflakes.Snowflake = attr.field(eq=False, hash=False, repr=True)
-    """The channel ID this webhook is for."""
-
-    author: typing.Optional[users_.User] = attr.field(eq=False, hash=False, repr=True)
-    """The user that created the webhook
-
-    !!! info
-        This will be `builtins.None` when getting a webhook with bot authorization rather
-        than the webhook's token.
-    """
-
-    name: typing.Optional[str] = attr.field(eq=False, hash=False, repr=True)
-    """The name of the webhook."""
-
-    avatar_hash: typing.Optional[str] = attr.field(eq=False, hash=False, repr=False)
-    """The avatar hash of the webhook."""
-
-    token: typing.Optional[str] = attr.field(eq=False, hash=False, repr=False)
-    """The token for the webhook.
-
-    !!! info
-        This is only available for incoming webhooks that are created in the
-        channel settings.
-    """
-
-    application_id: typing.Optional[snowflakes.Snowflake] = attr.field(eq=False, hash=False, repr=False)
-    """The ID of the application that created this webhook."""
-
-    source_channel: typing.Optional[channels_.PartialChannel] = attr.field(eq=False, hash=False, repr=True)
-    """The partial object of the channel a `CHANNEL_FOLLOWER` webhook is following.
+class ExecutableWebhook(abc.ABC):
+    """An abstract class with logic for executing entities as webhooks."""
 
-    Will be `builtins.None` for other webhook types.
-    """
+    # This is a mixin, do not add slotted fields.
+    __slots__: typing.Sequence[str] = ()
 
-    source_guild: typing.Optional[guilds_.PartialGuild] = attr.field(eq=False, hash=False, repr=True)
-    """The partial object of the guild a `CHANNEL_FOLLOWER` webhook is following.
-
-    Will be `builtins.None` for other webhook types.
-    """
+    @property
+    @abc.abstractmethod
+    def app(self) -> traits.RESTAware:
+        """Client application that models may use for procedures.
 
-    def __str__(self) -> str:
-        return self.name if self.name is not None else f"Unnamed webhook ID {self.id}"
+        Returns
+        -------
+        hikari.traits.RESTAware
+            The client application that models may use for procedures.
+        """
 
     @property
-    def mention(self) -> str:
-        """Return a raw mention string for the given webhook's user.
+    @abc.abstractmethod
+    def webhook_id(self) -> snowflakes.Snowflake:
+        """ID used to execute this entity as a webhook.
 
-        !!! note
-            This exists purely for consistency. Webhooks do not receive events
-            from the gateway, and without some bot backend to support it, will
-            not be able to detect mentions of their webhook.
-
-        Example
+        Returns
         -------
+        hikari.snowflakes.Snowflake
+            The ID used to execute this entity as a webhook.
+        """
```

Comment:
In what case could this be different to `.id`?

Reply:
For interactions (?)